src/WallEvaluatorColloid.h

```diff
@@ -52,8 +60,52 @@ namespace detail
 class WallEvaluatorColloid
     {
     public:
-    //! Define the parameter type used by this wall potential evaluator
-    typedef Scalar2 param_type;
+    struct param_type
+        {
+        DEVICE void load_shared(char*& ptr, unsigned int& available_bytes) { }
+
+        HOSTDEVICE void allocate_shared(char*& ptr, unsigned int& available_bytes) const { }
+
+#ifndef ENABLE_HIP
+        //! set CUDA memory hints
+        void set_memory_hint() const { }
+#endif
+
+#ifndef __HIPCC__
+        param_type() : sigma(0), epsilon(0), a(0), lj1(0), lj2(0) { }
+
+        param_type(pybind11::dict v, bool managed = false)
+            {
+            sigma = v["sigma"].cast<Scalar>();
+            epsilon = v["epsilon"].cast<Scalar>();
+            a = v["a"].cast<Scalar>();
+            // a = a * Scalar(0.5);
+            Scalar sigma_3 = sigma * sigma * sigma;
+            Scalar sigma_6 = sigma_3 * sigma_3;
+            lj1 = epsilon * sigma_6 / Scalar(7560.0);
+            lj2 = epsilon / Scalar(6.0);
+            }
+
+        pybind11::dict asDict()
+            {
+            pybind11::dict v;
+            v["sigma"] = sigma;
+            v["epsilon"] = epsilon;
+            v["a"] = a;
+            return v;
+            }
+#endif
+        Scalar sigma;
+        Scalar epsilon;
+        Scalar a;
+        Scalar lj1;
+        Scalar lj2;
+        }
+#if HOOMD_LONGREAL_SIZE == 32
+        __attribute__((aligned(8)));
+#else
+        __attribute__((aligned(16)));
+#endif
```

Comment:
Align these to Scalar4 boundaries, since there are 5 parameters.
```suggestion
#if HOOMD_LONGREAL_SIZE == 32
        __attribute__((aligned(16)));
#else
        __attribute__((aligned(32)));
#endif
```